core/test/network-records-to-devtools-log.js

```diff
@@ -40,20 +46,181 @@ function headersArrayToHeadersDict(headersArray = []) {
 }
 
 /**
+ * Returns true if the time is defined, false if the time is `undefined` (not
+ * provided) or `-1` (default prop value in `NetworkRequest`).
+ * @param {number|undefined} time
+ * @return {boolean}
+ */
+function timeDefined(time) {
+  return time !== undefined && time !== -1;
+}
+
+/**
+ * Asserts that any value in the object is less than or equal to any later
+ * values (in object enumeration order). `undefined` or `-1` values are ignored.
+ * Keys of the object are used for better error logging.
+ * @param {Record<string, number|undefined>} values
+ */
+function assertTimingIncreases(values) {
+  const keys = Object.keys(values);
+  for (let i = 0; i < keys.length - 1; i++) {
+    const step = keys[i];
+    if (!timeDefined(values[step])) continue;
+
+    for (let j = i + 1; j < keys.length; j++) {
+      const comparison = keys[j];
+      if (!timeDefined(values[comparison])) continue;
+      assert(values[step] <= values[comparison], `'${step}' (${values[step]}) exceeds '${comparison}' (${values[comparison]}) in test network record`); // eslint-disable-line max-len
```

Comment:
I had a minor concern about how much this would generate the error string without ever needing to show it, but this path is hit <3000 times when running the core tests so not a big deal.